cmd/ocsp-updater/main.go

```diff
@@ -542,19 +550,29 @@ const clientName = "OCSP"
 func setupClients(c cmd.OCSPUpdaterConfig, stats metrics.Statter) (
 	core.CertificateAuthority,
 	core.Publisher,
+	pubPB.PublisherClient,
+	time.Duration,
 	core.StorageAuthority,
 ) {
 	amqpConf := c.AMQP
 	cac, err := rpc.NewCertificateAuthorityClient(clientName, amqpConf, stats)
 	cmd.FailOnError(err, "Unable to create CA client")
 
-	pubc, err := rpc.NewPublisherClient(clientName, amqpConf, stats)
-	cmd.FailOnError(err, "Unable to create Publisher client")
-
+	var pubc core.Publisher
+	var grpcPubc pubPB.PublisherClient
+	var grpcTimeout time.Duration
+	if c.Publisher != nil {
+		conn, err := bgrpc.ClientSetup(c.Publisher)
+		cmd.FailOnError(err, "Failed to load credentials and create connection to service")
+		grpcPubc = pubPB.NewPublisherClient(conn)
+		grpcTimeout = c.Publisher.Timeout.Duration
+	} else {
+		pubc, err = rpc.NewPublisherClient(clientName, amqpConf, stats)
+		cmd.FailOnError(err, "Unable to create Publisher client")
+	}
 	sac, err := rpc.NewStorageAuthorityClient(clientName, amqpConf, stats)
 	cmd.FailOnError(err, "Unable to create SA client")
-
-	return cac, pubc, sac
+	return cac, pubc, grpcPubc, grpcTimeout, sac
```

Comment:
Again, this would all be much cleaner if pubc and grpcPubc were the same variable.